httpbakery/visitor.go

```diff
@@ -0,0 +1,108 @@
+package httpbakery
+
+import (
+	"net/http"
+	"net/url"
+
+	"github.com/juju/httprequest"
+	"gopkg.in/errgo.v1"
+)
+
+var ErrMethodNotSupported = errgo.New("interaction method not supported")
+
+// Visitor represents a handler that can handle ErrInteractionRequired errors
+// from a client's discharge request. The methodURLs parameter to
+// VisitWebPage holds a set of possible ways to complete the
+// discharge request. When called directly from Client, this
+// will contain only a single entry with the UserInteractionMethod key,
+// specifying that the associated URL should be opened
+// in a web browser for the user to interact with.
+//
+// See FallbackVisitor for a way to gain access to
+// alternative methods.
+//
+// A Visitor implementation may return ErrMethodNotSupported
+// if it cannot handle any of the supplied methods.
+type Visitor interface {
+	VisitWebPage(client *Client, methodURLs map[string]*url.URL) error
```

Comment:
may i propose a new name for this method: DoInteraction.. i think this has outgrown it's original purpose - to use a web browser to open a web page..